src/main/java/org/opensearch/neuralsearch/processor/chunker/ChunkerParameterParser.java

```diff
@@ -39,14 +39,36 @@ public static String parseStringParameter(final Map<String, Object> parameters,
         return fieldValue.toString();
     }
 
+    /**
+     * Parse integer type parameter without default value
+     * Throw IllegalArgumentException if parameter is missing or is not an integer.
+     */
+    public static int parseIntegerParameter(final Map<String, Object> parameters, final String fieldName) {
+        if (!parameters.containsKey(fieldName)) {
```

Comment:
```suggestion
       return parseIntegerParameter(parameters, fieldName, null);
```

Reply:
I think this design doesn't follow conventions in other place in Java API.
If you calling parse(argument) than it should execute parsing logic
If you calling parse(argument, defaultValue) than it should try to execute parsing logic, and return default value in case that parsing logic return empty value

In your code everywhere where this new method called with null you can call your new method without default value.
And in that new method implementation you can just call this method with defaultValue = null, see my previous comment with suggestion

Reply:
Oh, I don't think the function is necessary. I have removed it.

Reply:
> If you calling parse(argument, defaultValue) than it should try to execute parsing logic, and return default value in case that parsing logic return empty value

Not very sure what you mean here. We should throw illegal argument exception if necessary.

Reply:
> In your code everywhere where this new method called with null you can call your new method without default value.

Updated by Zan's comment. You can review now.